lib/types.nix

```diff
@@ -107,22 +107,83 @@ rec {
       merge = mergeEqualOption;
     };
 
-    int = mkOptionType rec {
-      name = "int";
-      description = "integer";
-      check = isInt;
-      merge = mergeOneOption;
-    };
+### INTEGERS
+
+    int = ints.signed;
+
+    # specialized subdomains of int
+    ints =
+      let
+        int = mkOptionType rec {
+          name = "int";
+          description = "signed integer";
+          check = isInt;
+          merge = mergeOneOption;
+        };
+
+        betweenDesc = lowest: highest:
+          "${toString lowest} and ${toString highest} (both inclusive)";
+        between = lowest: highest: assert lowest <= highest;
+          addCheck int (x: x >= lowest && x <= highest) // {
+            name = "intBetween";
+            description = "Integer between ${betweenDesc lowest highest}.";
+          };
+        ign = lowest: highest: name: docStart:
```

Comment:
what does ign mean?

Reply:
Helper w/ cute name only used internally in `sign` and `unsign` directly below.